console/src/hooks/useUndoableDispatch.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+import { type Dispatch, type Selector, type UnknownAction } from "@reduxjs/toolkit";
+import { deep } from "@synnaxlabs/x";
+import { useCallback, useRef } from "react";
+import { useStore } from "react-redux";
+
+interface History<T> {
+  past: T[];
+  present: T | null;
+  future: T[];
+}
+
+export const useUndoableDispatch = <
```

Comment:
Should be able to configure the undo stack size. Also set it to a reasonable amount, as this hook could cause a memory leak with a large undo history.